tests/functional/event_handler/test_api_gateway.py

```diff
@@ -1818,110 +1816,32 @@ def get_lambda(my_id: str, name: str, injected: str) -> Response:
-
-    @app.get("/without-cors")
-    def without_cors() -> Response:
-        return Response(200, content_types.TEXT_HTML, "test")
-
-    def handler(event, context):
-        return app.resolve(event, context)
-
-    # WHEN calling the event handler
-    result = handler(LOAD_GW_EVENT, None)
-
-    # THEN the headers should include cors headers
-    assert "multiValueHeaders" in result
-    headers = result["multiValueHeaders"]
-    assert headers["Content-Type"] == [content_types.TEXT_HTML]
-    assert headers["Access-Control-Allow-Origin"] == ["https://aws.amazon.com"]
-    assert "Access-Control-Allow-Credentials" not in headers
-    assert headers["Access-Control-Allow-Headers"] == [",".join(sorted(CORSConfig._REQUIRED_HEADERS))]
-
-    # THEN for routes without cors flag return no cors headers
-    mock_event = {"path": "/my/request", "httpMethod": "GET"}
-    result = handler(mock_event, None)
-    assert "Access-Control-Allow-Origin" not in result["multiValueHeaders"]
-
-
-def test_router_cors_middleware():
-    # GIVEN a function with cors middleware (class definition)
-    # AND http method set to GET
-    app = ApiGatewayResolver()
-    app.use(CORSMiddleware())
-
-    @app.get("/my/path")
-    def with_cors() -> Response:
-        return Response(200, content_types.TEXT_HTML, "test")
-
-    @app.get("/without-cors")
-    def without_cors() -> Response:
-        return Response(200, content_types.TEXT_HTML, "test")
-
-    def handler(event, context):
-        return app.resolve(event, context)
-
-    # WHEN calling the event handler
-    result = handler(LOAD_GW_EVENT, None)
-
-    # THEN the headers should include cors headers
-    assert "multiValueHeaders" in result
-    headers = result["multiValueHeaders"]
-    assert headers["Content-Type"] == [content_types.TEXT_HTML]
-    assert headers["Access-Control-Allow-Origin"] == ["https://aws.amazon.com"]
-    assert "Access-Control-Allow-Credentials" not in headers
-    assert headers["Access-Control-Allow-Headers"] == [",".join(sorted(CORSConfig._REQUIRED_HEADERS))]
-
-    # THEN for routes without cors flag return no cors headers
-    mock_event = {"path": "/my/request", "httpMethod": "GET"}
-    result = handler(mock_event, None)
-    assert "Access-Control-Allow-Origin" not in result["multiValueHeaders"]
-
-
-def test_cache_control_middleware_200():
-    # GIVEN a function with cache_control set
+def test_middleware_early_return():
+    # GIVEN
     app = ApiGatewayResolver()
 
-    @app.get("/success", middleware=[CacheControlMiddleware("max-age=600")])
-    def with_cache_control() -> Response:
-        return Response(200, content_types.TEXT_HTML, "has 200 response")
-
-    def handler(event, context):
-        return app.resolve(event, context)
-
-    # WHEN calling the event handler
-    # AND the function returns a 200 status code
-    result = handler({"path": "/success", "httpMethod": "GET"}, None)
+    def middleware_one(app, get_response, **kwargs):
```

Comment:
```suggestion
    def middleware_one(app, get_response, **context):
```

having as `**context` will be a good thing for docs later.